crates/goose-cli/src/session/input.rs

```diff
@@ -588,4 +602,49 @@ mod tests {
             }
         }
     }
+
+    #[test]
+    #[serial]
+    fn test_get_newline_key_default() {
+        // Clear the env var to test default behavior
+        std::env::remove_var("GOOSE_CLI_NEWLINE_KEY");
+        assert_eq!(get_newline_key(), 'j');
+    }
+
+    #[test]
+    #[serial]
+    fn test_get_newline_key_custom() {
+        // Test setting a custom key
+        std::env::set_var("GOOSE_CLI_NEWLINE_KEY", "o");
+        assert_eq!(get_newline_key(), 'o');
+
+        // Test uppercase is converted to lowercase
+        std::env::set_var("GOOSE_CLI_NEWLINE_KEY", "N");
+        assert_eq!(get_newline_key(), 'n');
+
+        // Clean up
+        std::env::remove_var("GOOSE_CLI_NEWLINE_KEY");
+    }
+
+    #[test]
+    #[serial]
+    fn test_get_newline_key_empty_string() {
+        // Test empty string falls back to default
+        std::env::set_var("GOOSE_CLI_NEWLINE_KEY", "");
+        assert_eq!(get_newline_key(), 'j');
+
+        // Clean up
+        std::env::remove_var("GOOSE_CLI_NEWLINE_KEY");
+    }
+
+    #[test]
+    #[serial]
+    fn test_get_newline_key_first_char_only() {
+        // Test only first character is used
+        std::env::set_var("GOOSE_CLI_NEWLINE_KEY", "abc");
+        assert_eq!(get_newline_key(), 'a');
+
+        // Clean up
+        std::env::remove_var("GOOSE_CLI_NEWLINE_KEY");
+    }
```

Comment:
actually just delete these tests, I don't think they add much (https://engineering.block.xyz/blog/the-high-cost-of-free-testing)